### PR TITLE
Added model creation helpers

### PR DESCRIPTION
**Pull request by subutai**
Added model factory
Added register_regions
Updated example scripts to use the factory

Some minor changes to model constructors to better support model_factory.

### CODE REVIEW

**Comment by subutai** [comment]
@BoltzmannBrain Ready for review.

File: htmresearch/frameworks/nlp/model_factory.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2015, Numenta, Inc.  Unless you have purchased from
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/model_factory.py]: 2016 :smile:

File: htmresearch/frameworks/nlp/model_factory.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+def getNetworkConfig(networkConfigPath):
+  """
+  Given path to JSON model config file, return a dict.
+  """
+  try:
+    with open(networkConfigPath, "rb") as fin:
+      return simplejson.load(fin)
+  except IOError as e:
+    print "Could not find network configuration JSON at \'{}\'.".format(
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/model_factory.py]: don't need escape characters for the single quotes

**Comment by BoltzmannBrain** [comment]
:+1:

**Pull request merged**